packages/sdk/src/store/store.ts

```diff
@@ -257,32 +264,81 @@ export class Store implements IStore {
     };
   }
 
-  private async getPeerToUse(): Promise<PeerId | undefined> {
-    let peerId: PeerId | undefined;
+  private async getPeerToUse(pubsubTopic: string): Promise<PeerId | undefined> {
+    const peers = await this.filterConnectedPeers(pubsubTopic);
+
+    const peer = this.options.peers
+      ? await this.getPeerFromConfigurationOrFirst(peers, this.options.peers)
+      : peers[0]?.id;
+
+    return peer;
+  }
 
-    if (this.options?.peer) {
-      const connectedPeers = await this.connectionManager.getConnectedPeers();
+  private async getPeerFromConfigurationOrFirst(
+    peers: Peer[],
+    configPeers: string[]
+  ): Promise<PeerId | undefined> {
+    const storePeers = configPeers.map(multiaddr);
+    const missing = [];
 
-      const peer = connectedPeers.find(
-        (p) => p.id.toString() === this.options?.peer
+    for (const peer of storePeers) {
```

Comment:
nit: `storePeers` is confusing. is it a peer that supports store protocol or a list of peers from the current peer store?